Source: http://www.mail-archive.com/screen-devel@gnu.org/msg00556.html
Upstream: http://savannah.gnu.org/bugs/?30880
Reason: termcap entries longer than 20 chars cause
        screen to fail to execute ('rxvt-unicode-256color', in particular)

diff --git a/display.h b/display.h
index ef99954..064c4c5 100644
--- a/display.h
+++ b/display.h
@@ -85,7 +85,7 @@ struct display
   struct win *d_other;		/* pointer to other window */
   int   d_nonblock;		/* -1 don't block if obufmax reached */
 				/* >0: block after nonblock secs */
-  char  d_termname[20 + 1];	/* $TERM */
+  char  d_termname[31 + 1];	/* $TERM */
   char	*d_tentry;		/* buffer for tgetstr */
   char	d_tcinited;		/* termcap inited flag */
   int	d_width, d_height;	/* width/height of the screen */

diff --git a/screen.h b/screen.h
index 5765d56..1a46024 100644
--- a/screen.h
+++ b/screen.h
@@ -198,7 +198,7 @@ struct msg
 	  char preselect[20];
 	  int esc;		/* his new escape character unless -1 */
 	  int meta_esc;		/* his new meta esc character unless -1 */
-	  char envterm[20 + 1];	/* terminal type */
+	  char envterm[31 + 1];	/* terminal type */
 	  int encoding;		/* encoding of display */
 	}
       attach;
-- 
1.7.2.3

